### PR TITLE
[IMP] Stock: Improved return mechanism for moves [ADD] Added a field ret...

### PR DESCRIPTION
**Pull request by ara-odoo**
...urn_location to specify return location in return wizard.
Task: https://www.odoo.com/web?#id=10435&view_type=form&model=project.task&menu_id=3942&action=327 
Pad:https://pad.odoo.com/p/r.HuLGGOK8cxkKMBVj

### CODE REVIEW

File: addons/stock/wizard/stock_return_picking.py [hunk 1]
```diff
@@ -134,6 +135,8 @@ def _create_returns(self, cr, uid, ids, context=None):
             new_qty = data_get.quantity
             if new_qty:
                 returned_lines += 1
+                location = self.browse(cr, uid, ids, context=context).location_id
```
Comment (ara-odoo) [inline addons/stock/wizard/stock_return_picking.py]: no need to browse record, you can get location info from wizard data?

File: addons/stock/wizard/stock_return_picking.py [hunk 2]
```diff
@@ -134,6 +135,8 @@ def _create_returns(self, cr, uid, ids, context=None):
...
+                location_id = location.id if location.id else move.location_id.id
                 move_obj.copy(cr, uid, move.id, {
```
Comment (ara-odoo) [inline addons/stock/wizard/stock_return_picking.py]: location_id = location and location.id or move.location_id.id??

File: addons/stock/wizard/stock_return_picking.py [hunk 3]
```diff
@@ -43,7 +44,33 @@ class stock_return_picking(osv.osv_memory):
...
         'move_dest_exists': fields.boolean('Chained Move Exists', readonly=True, help="Technical field used to hide help tooltip if not needed"),
+        'location_id': fields.many2one('stock.location', 'Return Location'),
     }
+  
+    def fields_view_get(self, cr, uid, view_id=None, view_type=False, context=None, toolbar=False, submenu=False):
+        """
+            Add domain 'allow_check_writting = True' on journal_id field and remove 'widget = selection' on the same
+            field because the dynamic domain is not allowed on such widget
+        """
+
```
Comment (ara-odoo) [inline addons/stock/wizard/stock_return_picking.py]: remove doc string i thing its not related to return location.

File: addons/stock/wizard/stock_return_picking.py [hunk 4]
```diff
@@ -43,7 +44,33 @@ class stock_return_picking(osv.osv_memory):
...
+        if not context: context = {}
+        res = super(stock_return_picking, self).fields_view_get(cr, uid, view_id=view_id, view_type=view_type, context=context, toolbar=toolbar, submenu=submenu)
+        record_id = context.get('active_id')
```
Comment (ara-odoo) [inline addons/stock/wizard/stock_return_picking.py]: picking_id?

**Comment by hmo-odoo** [comment]
continue on PR: https://github.com/odoo/odoo/pull/7337

**Pull request closed without merging**